From 550f3448650d86fdfdc8457226c505f026822279 Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Wed, 27 Nov 2013 15:59:23 +0800
Subject: [PATCH 18/23] ENGR00289842 fix R and B color reverse issue on
 mx6sl_evk

display color not correct, R and B reverse on KitKat
root cause: Google hard code WindowSurface format to RGBA_8888
for 32bpp, fix by remove such hard code so that BGRA_8888 can
also be supported.

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 opengl/libs/EGL/eglApi.cpp | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/opengl/libs/EGL/eglApi.cpp b/opengl/libs/EGL/eglApi.cpp
index ce92fb8..468c96f 100644
--- a/opengl/libs/EGL/eglApi.cpp
+++ b/opengl/libs/EGL/eglApi.cpp
@@ -457,8 +457,13 @@ EGLSurface eglCreateWindowSurface(  EGLDisplay dpy, EGLConfig config,
         EGLint a = 0;
         cnx->egl.eglGetConfigAttrib(iDpy, config, EGL_ALPHA_SIZE, &a);
         if (a > 0) {
-            // alpha-channel requested, there's really only one suitable format
-            format = HAL_PIXEL_FORMAT_RGBA_8888;
+            // remove hard code to RGBA_8888, need support BGRA format also,
+            // get real foramt by visual id, if fail then hard code to RGBA.
+            if (!cnx->egl.eglGetConfigAttrib(iDpy,
+                    config, EGL_NATIVE_VISUAL_ID, &format)) {
+                // alpha-channel requested, there's really only one suitable format
+                format = HAL_PIXEL_FORMAT_RGBA_8888;
+            }
         } else {
             EGLint r, g, b;
             r = g = b = 0;
-- 
1.8.0

